File: amuta/amuta-crm/Seldat.Amuta.Web/src/app/students/shared/models/branch-student.ts
import { Student } from './student.model';
import { Branch } from 'src/app/branches/shared/models/branch.model';
import { StudyPath } from 'src/app/branches/shared/models/study-path';

export class BranchStudent {
    constructor(
        //public Id:number=0,       
        public Student: Student=null ,
        public Branch :Branch=null,
        public EntryHebrewDate:Date=null ,
        public EntryGregorianDate:Date=null ,
        public ReleaseHebrewDate:Date=null,
        public ReleaseGregorianDate:Date=null ,
        public StudyPath :StudyPath=null ,
        public Status:number=0,
        public IsActive:boolean=false,
        public IsHead:boolean=false
    ){}

}

File: amuta/amuta-crm/Seldat.Amuta.Entities/BranchStudent.cs
using System;
using System.ComponentModel.DataAnnotations;


namespace Seldat.Amuta.Entities
{
    public enum StatusInBranch
    {
        Pending=1,
        Approved=2
    }
    public class BranchStudent
    {  
        public int Id { get; set; }
       // [Required]
        public Student Student { get; set; }
       // [Required]
        public Branch Branch { get; set; }
        public DateTime EntryHebrewDate { get; set; }
        public DateTime EntryGregorianDate { get; set; }
        public DateTime ReleaseHebrewDate { get; set; }
        public DateTime ReleaseGregorianDate { get; set; }
        public StudyPath StudyPath { get; set; }
       // [Required]
        public StatusInBranch Status { get; set; }
       // [Required]
        public bool IsActive { get; set; }
    }
}
